### PR TITLE
Add new lifecycle method: oninsert.

### PR DESCRIPTION
**Pull request by jbucaran**
- oninsert is a new lifecycle method that occurs after and
  element is created and mounted/inserted into the DOM.

- Also rename lifecycle events -> lifecycle methods to avoid
  creating new/unfamiliar terms.

- Update docs.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -106,6 +106,8 @@ export function app(app) {
       for (var i in node.data) {
         if (i === "oncreate") {
           node.data[i](element)
+        } else if (i === "onmount") {
```
Comment (zaceno) [inline src/app.js]: `oninsert`... right?

File: test/lifecycle.test.js [hunk 2]
```diff
@@ -3,13 +3,31 @@ import { expectHTMLToBe } from "./util"
...
+  })
+})
+
+test("onmount", done => {
+  app({
+    view: () =>
       h("div", {
-        oncreate: e => {
-          expect(state).toBe(1)
+        onmount: element => {
```
Comment (zaceno) [inline test/lifecycle.test.js]: `oninsert` ...

File: test/lifecycle.test.js [hunk 3]
```diff
@@ -21,7 +39,7 @@ test("onupdate", done => {
     state: 1,
     view: state =>
       h("div", {
-        onupdate: e => {
+        onupdate: element => {
```
Comment (zaceno) [inline test/lifecycle.test.js]: `oninsert`...

Comment (zaceno) [inline test/lifecycle.test.js]: nevermind. ... not sure why I flagged this one :P

File: src/app.js (hunk 1 above)
Comment (jbucaran) [inline src/app.js]: lol

**Comment by jbucaran** [comment]
@zaceno Fixed! 🎉

**Comment by zaceno** [comment]
... oh you already merged it. Thats why, Lol!

**Comment by jamen** [comment]
Hmm, I am unsure about "Lifecycle Methods" though.  What is the confusion?

**Comment by jbucaran** [comment]
@jamen I am borrowing the term from react. Newcomers are likely to be more familiar with that term. But more importantly, lifecycle events !== appOptions.events, which may actually cause confusion.

**Comment by jamen** [comment]
Agree on the last point.

The first point is trivial imo, but I'll just say this:  "methods" seems like object-orientated wording to me.  Because in React they are [methods on a class](https://facebook.github.io/react/docs/react-component.html#the-component-lifecycle). Here I just thought of them as functions and more specifically events.  Especially with the usage `<h1 oninsert={fn}>`.

**Comment by Swizz** [comment]
- Snabbdom call them (lifecycle events/methods) Hooks
- Preact is with Lifecycle methods too (It seem you are a lot inspired by Preact isnt it ?)
- Vue is in favor of Lifecycle Hooks
- Inferno use Lifecycle events

I think naming is not an issue ; if the doc is well written

**Comment by jbucaran** [comment]
> @jamen "methods" seems like object-orientated wording

Good point. Oh, what to do...? 🤔

**Comment by SkaterDad** [comment]
My vote would be "hooks" or "events".

**Comment by Swizz** [comment]
Hooks looks good associated to a Lifecycle diagram.

e.g : [Vue - Lifecycle Diagram](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram)
This diagram show us the instance lifecycle ; and in red there are some **hooks** to catch some moments of the instance lifecycle.

**Comment by jbucaran** [comment]
So, I went with events for everything. I like the harmony it brings to the docs.

- app `events` property.
- custom events, created via `emit("myCustomEvent", data)`
- lifecycle events, oncreate, oninsert, onupdate, onremove

**Comment by Swizz** [comment]
> - lifecycle events, oncreate, oninsert, onupdate, onremove

Lifecycle Events looks also great associated to DOM Events. Same API, one deal with Lifecycle, the other one with regular events : 

````jsx
<div
  oninsert={ () => {} }
  onupdate={ () => {} }
  onclick={ () => {} }
  onhover={ () => {} }
/>

**Pull request merged**